lib/envars.ml

```diff
@@ -200,21 +200,23 @@ let xdg_data_dirs warn =
 let xdg_dirs ~warn =
   List.filter Sys.file_exists (xdg_data_dirs warn)
 
+let esc s = if String.contains s ' ' then "\"" ^ s ^ "\"" else s
+
 (* Print the configuration information *)
 
 let print_config ?(prefix_var_name="") f coq_src_subdirs =
-  let open Printf in 
+  let open Printf in
   fprintf f "%sLOCAL=%s\n" prefix_var_name (if Coq_config.local then "1" else "0");
-  fprintf f "%sCOQLIB=%s/\n" prefix_var_name (coqlib ());
-  fprintf f "%sDOCDIR=%s/\n" prefix_var_name (docdir ());
-  fprintf f "%sOCAMLFIND=%s\n" prefix_var_name (ocamlfind ());
-  fprintf f "%sCAMLP4=%s\n" prefix_var_name Coq_config.camlp4;
-  fprintf f "%sCAMLP4O=%s\n" prefix_var_name Coq_config.camlp4o;
-  fprintf f "%sCAMLP4BIN=%s/\n" prefix_var_name (camlp4bin ());
-  fprintf f "%sCAMLP4LIB=%s\n" prefix_var_name (camlp4lib ());
+  fprintf f "%sCOQLIB=%s/\n" prefix_var_name (esc (coqlib ()));
```

Comment:
This is wrong in the context of `coq_makefile`. It uses this function to fill in `coq_makefile.conf`, that is in `make` syntax, not shell.
In that context `' '` is not a problem, and we already use `"` when the shell is invoked.  With this patch I'm afraid we would end up with extra `"` all over the place.

Example:
```
gares@gargamel:/tmp$ cat > Makefile
X = "foo bar" # this is what this patch generates (in the .conf file)
all:
	"$(X)" -baz  # this is alreay part of CoqMakefile.in

gares@gargamel:/tmp$ make
echo 

""foo bar"" -bar
/bin/sh: 1: foo: not found
```
As you can see, the double `"` end up being a no-op (the shell look for `foo` and not `foo bar`).

Reply:
Indeed, I learnt that. But finally, I also heard that `make` has escaping for `$`, so finally, I don't see what would prevent it in theory to support escaped spaces.

PR updated (but I don't know how to tell that changes are not anymore needed).

Reply:
> but I don't know how to tell that changes are not anymore needed

You can ask @gares to dismiss his review or do it yourself (dismiss review in the box at the bottom of the conversation).

Reply:
Side remark, how would you translate "dismiss" in French. My dictionary gives it a rather hard meaning (congédier, renvoyer, écarter, rejeter, rendre une fin de non-recevoir, 
ne pas prendre au sérieux, ...). I don't want to do anything of this kind to @gares' report. I just want to tell that I took its comments into consideration.

Reply:
I don't know French, but the English definition that I'd use here is "send away".  Like if you have a bunch of notifications, and you want to not see them or make them disappear, you can dismiss them, that is, send them away.  I think the connotation I have here is "it's no longer needed" (though of course there are other harsher uses of "dismiss" that suggest invalidating something).

Reply:
From Google translate, I'd probably go with congédier or faire partir (but, again, I don't know French)

Reply:
@JasonGross: "congédier" is more about someone, almost like "to fire"/"to sack" I guess.

So, the less "invalidating something" meaning is "no longer needed". Anyway, I'm surprised by the use of this word on github. Naively, I would have said that we can disagree with the request, or that we can approve (and then precisely validate it), and then click to inform that the request is granted.

Reply:
Yeah, I found "dismiss" slightly surprising, too.

I think the idea is that it's up to the reviewer whether or not an update meets the change request, and dismissing a request is neither for agreeing nor disagreeing, but for administrating.  [The GitHub docs](https://help.github.com/articles/dismissing-a-pull-request-review/) say:
> If your repository requires reviews, you can dismiss pull request reviews that are no longer valid or are unable to be approved by the reviewer.
>
> If a person who requested changes isn't available to give an approved review, or if a pull request has changed significantly since it was reviewed, repository administrators or people with write access can dismiss a review.
>
> This changes the status of the review to a review comment. When you dismiss a review, you must add a comment explaining why you dismissed it. Your comment will be added to the pull request conversation.

Reply:
Well, indeed, dismissing seems here to refer to "no longer valid". (At another [place](https://help.github.com/articles/reviewing-proposed-changes-in-a-pull-request/), converging to this meaning, it is also written "stale review dismissal".)

So, with this meaning in mind, and following @Zimmi48's words: @gares, I guess you shall agree with the changes, so would you be kind to dismiss your request?

Reply:
I'm assuming that @gares assumes that marking requests as no longer valid is the responsibility of the PR submitter, so "dismissing" myself.